### PR TITLE
✨ Plugin delete subcommand implementation

### LINKED ISSUE #396: Implement plugin delete functionality
**User Story**

As an operator user ] I would like to selectively cleanup operator installation from the cluster.

**Detailed Description**

Currently the `delete` subcommand structure is provided in the plugin, but it needs to be implemented.

**Anything else you would like to add:**

/kind feature

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
**What this PR does / why we need it**:
Depends on #334, implementing delete for provider manifests.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #396

### CODE REVIEW

File: cmd/plugin/cmd/delete.go [hunk 1]
```diff
@@ -98,20 +117,20 @@ func init() {
 	deleteCmd.Flags().BoolVar(&deleteOpts.includeCRDs, "include-crd", false,
 		"Forces the deletion of the provider's CRDs (and of all the related objects)")
 
-	deleteCmd.Flags().StringVar(&deleteOpts.coreProvider, "core", "",
-		"Core provider version (e.g. cluster-api:v1.1.5) to delete from the management cluster")
+	deleteCmd.Flags().BoolVar(&deleteOpts.coreProvider, "core", false,
+		"Core provider (e.g. cluster-api) to delete from the management cluster")
```
Comment (alexander-demicev) [inline cmd/plugin/cmd/delete.go:121]: Let's document later differences in arguments of the operator plugin and clusterctl.

**Review by alexander-demicev — APPROVED** [review]
/approve

**Review by Fedosin — COMMENTED** [review]
Could you also update Examples section for the delete command and specify that --core flag is boolean now. I.e.
capioperator delete --core cluster-api -> capioperator delete --core

File: cmd/plugin/cmd/delete.go [hunk 2]
```diff
@@ -138,9 +159,190 @@ func runDelete() error {
...
+			if err := client.Delete(ctx, ns); ctrlclient.IgnoreNotFound(err) != nil {
+				return false, fmt.Errorf("unable to issue delete for Namespace %s: %w", provider.GetNamespace(), err)
+			}
+		}
+	}
+
+	if len(providerList.GetItems()) > 0 {
+		log.Info(fmt.Sprintf("%d items remaning...", len(providerList.GetItems())))
+		return false, nil
+	}
```
Comment (Fedosin) [inline cmd/plugin/cmd/delete.go:335]: I suggest to add 
else {
    log.Info("All required providers were deleted")
}

**Comment by Fedosin** [comment]
When I execute `capioperator delete --all --include-crd`, it also deletes operator CRDs. Probably we don't want it.

**Comment by Fedosin** [comment]
For some reason if there are no providers installed, this command deletes CRDs, but not the deployment itself:
```sh
➭ ./bin/clusterctl-operator init -i "" --core "" -b "" -c ""
Checking that Cert Manager is installed and running.
Skipping installing cert-manager as it is already installed
Checking that CAPI Operator is installed and running.
Installing CAPI operator Version="latest"
Detected latest operator version Version="v0.8.1"
Waiting for CAPI Operator to be available...
CAPI Operator is successfully installed.
➭ ./bin/clusterctl-operator delete --all --include-crd  --include-namespace
Waiting for CAPI Operator manifests to be removed...
➭ k get deploy -n capi-operator-system         
NAME                               READY   UP-TO-DATE   AVAILABLE   AGE
capi-operator-controller-manager   1/1     1            1           78s
```

**Comment by Danil-Grigorev** [comment]
> For some reason if there are no providers installed, this command deletes CRDs, but not the deployment itself:
> 
> ```shell
> ➭ ./bin/clusterctl-operator init -i "" --core "" -b "" -c ""
> Checking that Cert Manager is installed and running.
> Skipping installing cert-manager as it is already installed
> Checking that CAPI Operator is installed and running.
> Installing CAPI operator Version="latest"
> Detected latest operator version Version="v0.8.1"
> Waiting for CAPI Operator to be available...
> CAPI Operator is successfully installed.
> ➭ ./bin/clusterctl-operator delete --all --include-crd  --include-namespace
> Waiting for CAPI Operator manifests to be removed...
> ➭ k get deploy -n capi-operator-system         
> NAME                               READY   UP-TO-DATE   AVAILABLE   AGE
> capi-operator-controller-manager   1/1     1            1           78s
> ```

If there are no providers, then there are no known namespaces to delete. Is there some default pattern for namespace names in operator? `capi-operator-system` may belong to`--all` but this would look like side effect. CRDs however are always known and have no relation to deployment, so they are cleaned up.

File: cmd/plugin/cmd/delete.go (hunk 2 above)
Comment (Danil-Grigorev) [reply cmd/plugin/cmd/delete.go:335]: Added below.

**Review by Fedosin — COMMENTED** [review]
/retest

**Review by Fedosin — COMMENTED** [review]
/retest

**Review by Fedosin — APPROVED** [review]
/lgtm

**Pull request merged**